src/main/java/org/carlmontrobotics/subsystems/IntakeShooter.java

```diff
@@ -4,47 +4,120 @@
-	
-		CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(4, MotorConfig.NEO_550);
-//	CANSparkMax rightIntakeMotor = MotorControllerFactory.createSparkMax(5, MotorConfig.NEO_550);
-	
-		public IntakeShooter() {
-    }
-	
-		public void intake() {}//run both intake and passing motors
-    public void fire() {}//run both passing and firing motors
+    CANSparkMax leftShootingMotor = MotorControllerFactory.createSparkMax(leftShootingPort, MotorConfig.NEO);
+    CANSparkMax rightShootingMotor = MotorControllerFactory.createSparkMax(rightShootingPort, MotorConfig.NEO);
+    CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(intakePort, MotorConfig.NEO_550);
+
+	private final RelativeEncoder leftShootingEncoder = leftShootingMotor.getEncoder();
+	private final RelativeEncoder rightShootingEncoder = rightShootingMotor.getEncoder();
+
+	private final RelativeEncoder intakeMotorEncoder = intakeMotor.getEncoder();
+	//todo: figure out way to make it so its not "constants."
+	private final SparkPIDController leftShootingMotorPID = leftShootingMotor.getPIDController();
+	private final SparkPIDController rightShootingMotorPID = rightShootingMotor.getPIDController();
+	private final TimeOfFlight distSensor = new TimeOfFlight(distSensorPort);
+
+	//private final DigitalOutput beamBreak = new DigitalOutput(beamBreakPort);
+	//We may use this
+
+		//NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+		//NetworkTableEntry tx = table.getEntry("tx");
+		//NetworkTableEntry ty = table.getEntry("ty");
+		//NetworkTableEntry ta = table.getEntry("ta");
+
+		//SmartDashboard.putNumber("LimelightX", x);
+		//SmartDashboard.putNumber("LimelightY", y);
+		//SmartDashboard.putNumber("LimelightArea", area);
+
+		//the above commented out code is for limelight
+								
+	public IntakeShooter() {}
+
+	public void setRPMIntake(){
+		//Intake
+	}
+    	
+    public void setRPMSpeaker(){
+		//method that gets the speaker motors up to speed
+	}
+
+	public void setRPMAmp(){
+		//method that gets the Shooter motors up to a speed to eject a note
+	}
+
+	public void setRPMEject(){
+		//method that runs the shooting and intake motors backwards
+	}
+	public boolean isHoldingNote(){
+		//return beamBreak.get(); use if using beambraker
 		
-		public void eject() {}//throw ring onto ground (run all motors in reverse)
-	
-	
-    public void isHoldingNote(){
-			// method to check if holding note (beambreaker / digital input)
-    }
-
-  
+		//use distance sensor to check if holding note
+		//we will use get range();
+
+		/*/we will use an if statement and see if the 
+		distance is between two certain numbers.  If the
+		distance is between the two certain numbers, this will return true/*/
+
+		//Placeholder
+		return true;
+	}
+	public double getDistanceIn(){
+		return 0;
+		//TODO: explain in comments
+	}
+
     
+
     @Override
     public void periodic() {
-			//is pid needed? for rpm speeds?
+			//pid needed for rpm speeds
```

Comment:
typically pid controllers is initialized in the constructor, you wouldn't want to initialize your controllers every 20 ms.